src/RandomEnchants.cpp

```diff
@@ -1,63 +1,139 @@
-class RandomEnchantsPlayer : public PlayerScript{
+bool RandomEnchantEnabled = true;
+bool RandomEnchantAnnounce = true;
+bool Crafted = 1;
+bool Looted = 1;
+bool QuestReward = 1;
+uint32 HighQuality = 5;
+uint32 LowQuality = 1;
+uint32 Chance1 = 60;
+uint32 Chance2 = 65;
+uint32 Chance3 = 70;
+
+class REConfig : public WorldScript
+{
 public:
+	REConfig() : WorldScript("REConfig") { }
+
+	void OnBeforeConfigLoad(bool reload) override
+	{
+		if (!reload) {
+			std::string conf_path = _CONF_DIR;
+			std::string cfg_file = conf_path + "/mod_randomenchants.conf";
+
+			std::string cfg_def_file = cfg_file + ".dist";
+			sConfigMgr->LoadMore(cfg_def_file.c_str());
+			sConfigMgr->LoadMore(cfg_file.c_str());
 
-    RandomEnchantsPlayer() : PlayerScript("RandomEnchantsPlayer") { }
+			// Load Configuration Settings
+			SetInitialWorldSettings();
+		}
+	}
+
+	// Load Configuration Settings
+	void SetInitialWorldSettings()
+	{
+		RandomEnchantEnabled = sConfigMgr->GetBoolDefault("RandomEnchants.Enabled", true);
+		RandomEnchantAnnounce = sConfigMgr->GetBoolDefault("RandomEnchants.Announce", true);
+		Crafted = sConfigMgr->GetBoolDefault("RandomEnchants.OnCreate", 1);
+		Looted = sConfigMgr->GetBoolDefault("RandomEnchants.OnLoot", 1);
+		QuestReward = sConfigMgr->GetBoolDefault("RandomEnchants.OnQuestReward", 1);
+		HighQuality = sConfigMgr->GetIntDefault("RandomEnchants.HighQuality", 1);
+		LowQuality = sConfigMgr->GetIntDefault("RandomEnchants.LowQuality", 1);
+		Chance1 = sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60);
+		Chance2 = sConfigMgr->GetIntDefault("RandomEnchants.Chance2", 65);
+		Chance3 = sConfigMgr->GetIntDefault("RandomEnchants.Chance3", 70);
+
+		// Sanitize
+		if (HighQuality > 5) { HighQuality = 5; }
+		
+	}
+};
+
+class RandomEnchantsPlayer : public PlayerScript {
+public:
+
+	RandomEnchantsPlayer() : PlayerScript("RandomEnchantsPlayer") { }
+
+	void OnLogin(Player* player)  override
+	{
+		if (sConfigMgr->GetBoolDefault("RandomEnchants.Announce", true)) {
+			ChatHandler(player->GetSession()).SendSysMessage("This server is running the |cff4CFF00RandomEnchants |rmodule.");
+		}
+	}
 
-    void OnLogin(Player* player) override {
-		if (sConfigMgr->GetBoolDefault("RandomEnchants.AnnounceOnLogin", true))
-            ChatHandler(player->GetSession()).SendSysMessage(sConfigMgr->GetStringDefault("RandomEnchants.OnLoginMessage", "This server is running a RandomEnchants Module.").c_str());
-    }
 	void OnLootItem(Player* player, Item* item, uint32 /*count*/, uint64 /*lootguid*/) override
 	{
-		if (sConfigMgr->GetBoolDefault("RandomEnchants.OnLoot", true))
+		if (RandomEnchantEnabled)
+		{
 			RollPossibleEnchant(player, item);
+		}
 	}
+
 	void OnCreateItem(Player* player, Item* item, uint32 /*count*/) override
 	{
-		if (sConfigMgr->GetBoolDefault("RandomEnchants.OnCreate", true))
-			RollPossibleEnchant(player, item);
+		if (RandomEnchantEnabled)
+		{
+			if (Crafted == true)
+			{
+				RollPossibleEnchant(player, item);
+			}
```

Comment:
just realized you use tabs instead of space and it looks very empty on github xD

Reply:
ah crap, not sure if it was like this on stygiancore or if i did this, i think i got this module from there originally and updated it , this will require me to rewrite it all :( , maybe i can get to that after christmas

Reply:
Or you can simply replace all the "	" with "    ". From tab indents to 4 spaces indents, simple 15 seconds fix